File: packages/asset-manager/ios/Plugin/Classes/Results/ReadResult.swift
import Foundation
import Capacitor

@objc public class ReadResult: NSObject, Result {
    let data: String

    init(data: String) {
        self.data = data
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["data"] = data
        return result as AnyObject
    }
}

File: packages/asset-manager/android/src/main/java/io/capawesome/capacitorjs/plugins/assetmanager/classes/results/ReadResult.java
package io.capawesome.capacitorjs.plugins.assetmanager.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.assetmanager.interfaces.Result;

public class ReadResult implements Result {

    @NonNull
    private String data;

    public ReadResult(@NonNull String data) {
        this.data = data;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("data", data);
        return result;
    }
}
